src/v/cluster/metrics_reporter.cc

```diff
@@ -101,6 +105,53 @@ address parse_url(const ss::sstring& url) {
     }
     return ret;
 }
+
+ss::sstring get_hostname() {
+    std::array<char, HOST_NAME_MAX> hostname{};
+    if (::gethostname(hostname.data(), hostname.size()) != 0) {
+        return {};
+    }
+
+    return hostname.data();
+}
+
+ss::sstring get_domainname() {
+    std::array<char, HOST_NAME_MAX> domainname{};
+    if (::getdomainname(domainname.data(), domainname.size()) != 0) {
+        return {};
+    }
+
+    return domainname.data();
+}
+
+std::vector<ss::sstring> get_fqdn(std::string_view hostname) {
+    struct addrinfo hints = {};
+    struct addrinfo* info = nullptr;
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_CANONNAME;
+
+    auto free_info = ss::defer([&info] {
+        if (info != nullptr) {
+            freeaddrinfo(info);
+        }
+    });
+
+    if (getaddrinfo(hostname.data(), nullptr, &hints, &info) != 0) {
+        return {};
+    }
+
+    std::vector<ss::sstring> rv;
+    for (auto p = info; p != nullptr; p = p->ai_next) {
+        if (p->ai_canonname != nullptr) {
+            rv.emplace_back(p->ai_canonname);
+        }
+    }
+
+    return rv;
+}
```

Comment:
nit: put these in an anonymous namespace?

Reply:
why?

Reply:
So they can't conflict with a method `details::get_hostname` defined in another translation unit? The same reason you would make them `static` but anonymous namespaces are preferred over static: https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc/use-anonymous-namespace.html

Reply:
the other internal functions are declared in the header so they can't have internal linkage - these are not declared in the header so they can have internal linkage.

Reply:
fair enough... i'm not awake

Reply:
LGTM